Classes/Hooks/ContentDefender/ColumnConfigurationManipulationHook.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+namespace B13\Container\Hooks\ContentDefender;
+
+/*
+ * This file is part of TYPO3 CMS-based extension "container" by b13.
+ *
+ * It is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License, either version 2
+ * of the License, or any later version.
+ */
+
+use B13\Container\Domain\Factory\ContainerFactory;
+use B13\Container\Domain\Factory\Exception;
+use B13\Container\Tca\Registry;
+use IchHabRecht\ContentDefender\BackendLayout\ColumnConfigurationManipulationInterface;
+use TYPO3\CMS\Backend\Utility\BackendUtility;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Core\Utility\MathUtility;
+
+class ColumnConfigurationManipulationHook implements ColumnConfigurationManipulationInterface
+{
+    /**
+     * @var Registry
+     */
+    protected $tcaRegistry;
+
+    /**
+     * @var ContainerFactory
+     */
+    protected $containerFactory;
+
+    /**
+     * @var DatahandlerStorage
+     */
+    protected $datahandlerStorage;
+
+    public function __construct(
+        ContainerFactory $containerFactory = null,
+        Registry $tcaRegistry = null,
+        DatahandlerStorage $datahandlerStorage = null
+    ) {
+        $this->containerFactory = $containerFactory ?? GeneralUtility::makeInstance(ContainerFactory::class);
+        $this->tcaRegistry = $tcaRegistry ?? GeneralUtility::makeInstance(Registry::class);
+        $this->datahandlerStorage = $datahandlerStorage ?? GeneralUtility::makeInstance(DatahandlerStorage::class);
+    }
+
+    public function manipulateConfiguration(array $configuration, int $colPos, $recordUid): array
+    {
+        $parent = $this->getParentUid($recordUid);
+        if ($parent === 0) {
+            return $configuration;
+        }
+        try {
+            $container = $this->containerFactory->buildContainer($parent);
+        } catch (Exception $e) {
+            // not a container
+            return $configuration;
+        }
+        $cType = $container->getCType();
+        $configuration = $this->tcaRegistry->getContentDefenderConfiguration($cType, $colPos);
+        return $configuration;
+    }
+
+    private function getParentUid($recordUid): int
+    {
+        $parent = 0;
+        if (!empty($recordUid) && MathUtility::canBeInterpretedAsInteger($recordUid)) {
+            if ($this->datahandlerStorage->hasMapping((int)$recordUid)) {
+                $parent = $this->datahandlerStorage->getMapping((int)$recordUid);
+            } else {
+                // TcaCTypeItems: edit record
+                $record = BackendUtility::getRecord('tt_content', $recordUid, 'tx_container_parent');
```

Comment:
`getRecord()` or `getRecordWSOL()` here?

Reply:
`getRecord()` looks fine for me, the `$recordUid` is already the `uid` of the WS-Record.